### ISSUE TITLE
adjacent-overload-signatures mistakenly reports static member

### ISSUE THREAD

**Issue by giladgray** [palantir/tslint#2139]
### Bug Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: (please choose one) CLI / gulp

#### TypeScript code being linted

```ts
class GridEntry<T> {
    public static key(i: number, j: number) {
        return `${i}_${j}`;
    }

    public constructor (
        public i: number,
        public j: number,
        public value: T,
    ) {}

    // this method produces the error.  there are two things here 
    // called `key` but they're certainly not overloaded (one being static)
    public get key() {
        return GridEntry.key(this.i, this.j);
    }
}
```

https://github.com/palantir/blueprint/blob/master/packages/table/preview/store.ts

with `tslint.json` configuration:

```json
// from tslint:latest
{
  "adjacent-overload-signatures": true
}
```

#### Actual behavior

`public get key()` produces warning "all `key` signatures should be adjacent"

#### Expected behavior

Since one is static (`GridEntry.key()`) and another is instance-level (`grid.key()`), the rule shouldn't even apply here since there is no overload.

This seems similar to #1652 but is specifically about this case where the rule is erroneously applied.

**Comment by andy-hanson** [comment]
I can't reproduce this in tslint `4.4.2`.
